pkg/storegateway/indexheader/stream_binary_reader.go

```diff
@@ -148,7 +174,87 @@ func newFileStreamBinaryReader(path string, postingOffsetsInMemSampling int, log
 		return nil, err
 	}
 
-	return r, nil
+	return r, err
+}
+
+// loadSparseFromDisk loads in sparse index-header from disk.
+func (r *StreamBinaryReader) loadSparseFromDisk(logger log.Logger, sparseHeadersPath string, sparseData []byte, postingOffsetsInMemSampling int) (err error) {
```

Comment:
Besides the log message, the path doesn't actually seem used here.

```suggestion
func (r *StreamBinaryReader) loadFromSparseIndexHeader(logger log.Logger, sparseData []byte, postingOffsetsInMemSampling int) (err error) {
```